extensions/opentelemetry-exporter-jaeger/runtime/src/main/java/io/quarkus/opentelemetry/exporter/jaeger/LateBoundBatchSpanProcessor.java

```diff
@@ -0,0 +1,112 @@
+package io.quarkus.opentelemetry.exporter.jaeger;
+
+import org.jboss.logging.Logger;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import io.opentelemetry.sdk.trace.export.BatchSpanProcessor;
+
+/**
+ * Class to facilitate a delay in when the worker thread inside {@link BatchSpanProcessor}
+ * is started, enabling Quarkus to instantiate a {@link io.opentelemetry.api.trace.TracerProvider}
+ * during static initialization and set a {@link BatchSpanProcessor} delegate during runtime initialization.
+ */
+public class LateBoundBatchSpanProcessor implements SpanProcessor {
+    private static final Logger log = Logger.getLogger(LateBoundBatchSpanProcessor.class);
+
+    private boolean warningLogged = false;
+    private BatchSpanProcessor delegate;
+
```

Comment:
Are there threading concerns here? Should anything be marked volatile?

Reply:
I don't believe there are threading concerns a new instance of this would be created if there were multiple usages